### DISCUSSION TITLE
Cycling between `--term-playing-msg='\n${filename}'` and `--really-quiet`, and cycling video mode at the same time

### DISCUSSION THREAD

**Discussion by Kabouik** [mpv-player/mpv#14557] [Q&A] [answered]
I would like to be able to swith between `--no-video` and `sixel` views using the key `_`. This kind of works but the sixel view gets broken on my machine when mpv keeps refreshing text in the sixel mode, which is the case if I set a custom `term-playing-msg`  as shown in the title.

This is for a script meant to be user agnostic, so I am trying to avoid having to ship an `input.conf`. 

@guidocella on IRC kindly showed me tricks that could be assembled using `cycle` to achieve almost what I wanted:

```
mpv /path/to/file --no-config --vo=sixel --no-video --msg-level=all=error,statusline=status \
  --input-conf='memory://_ cycle-values msg-level all=error,statusline=status all=error; cycle vid'
```

However, this is based on showing the statusline when in text mode, and hiding it when in sixel mode (to avoid breaking the sixel cover with text refreshes). Sadly, this does not show the filename like the `term-msg-playing` option in the title, and I couldn't find how to use that with `cycle` or `cycle-values`. I suppose cycling between that and `really-quiet` simultaneously with the text/sixel change would work too, but I don't know how to alter `term-playing-msg` with this method.

Any ideas?

**Comment by guidocella** [comment]
`cycle-values term-playing-msg "\n${filename}" ''`

**Comment by Kabouik** [comment]
Thank you. I have two issues with that unfortunately: 

- Trying it in a dummy `input.conf` file where I give it a keybind, and then loading it at runtime with with `--input.conf='/tmp/input.conf'`, it seems the text is only shown temporarily instead of being a static toggle.
- Trying to add it directly in the list of options and using `memory ://` without a real file, I'm bumping into double-quotes and single-quote clashes. On top of that, this list of options is also a variable `$mpvui` in the script and I run `mpv <stuff> $mpvui`, where the initial assignation of the variable also needs quotes: 

```
mpvui="--vo=$secondary_interface --no-video --input-conf='memory://_ cycle-values term-playing-msg "\n${filename}" ''; cycle vid'"
```

I've tried different combinations of escaping or single and double quotes, with no success so far (but I understand this goes into shell scripting territory, not mpv).

**Comment by guidocella** [comment, marked as answer]
`--term-playing-msg` is only printed at the start of playback, you want `--term-status-msg` for text that can be changed during playback.

**Comment by Kabouik** [comment]
Thank you! I got what I wanted with this:

```
text_status="'\${?pause==no:▶ }\${?pause==yes:⏸ }\${percent-pos}% [\${time-pos}/\${duration}] \${filename} \${?pause==yes:(paused)}'"
mpv_opts="--no-config --ao=pulse --watch-later-dir=$HOME/.local/state/mpv/watch_later --save-position-on-quit --script-opts=osc-visibility=always --msg-level=all=error,statusline=status --term-status-msg=$text_status"
dummy_text="'$(printf \\033)'"
mpvui="--vo=sixel --no-video --input-conf='memory://_ cycle-values term-status-msg $dummy_text ${text_status//\'/\"}; cycle vid'"
mpv $mpv_opts $mpvui
```

**Comment by guidocella** [comment]
You can use `${osd-sym-cc}` for the play/pause symbol.

**Reply by Kabouik** [reply]
Thanks. That does not seem to work for me in 0.37.0 or 0.38.0 though. No big deal though as long as I can achieve the same with the conditional properties, which might be compatible with older versions.

**Reply by guidocella** [reply]
It seems not to work in the terminal, only in a window.

**Comment by Kabouik** [comment]
Turns out I overlooked an issue with the status not being refreshed properly after having switched to sixel and back to text view.

This video shows it (the black screen is when I press `_`):

https://asciinema.org/a/fQ2ptq0phNf1OAKWnHRbb1dZV

The command I'm using:

```bash
mpv somefile.mp3 someotherfile.mp3 --no-config \
  --vo=sixel \
  --no-video \
  --msg-level=all=error,statusline=status \
  --term-status-msg='${playlist}\n${?pause==no:▶ }${?pause==yes:⏸ }${percent-pos}% [${time-pos}/${duration}] ${filename} ${?pause==yes:(paused)}' \
  --input-conf='memory://_ cycle-values term-status-msg "Sixel shown here but not supported by asciinema" "${playlist}\\n${?pause==no:▶ }${?pause==yes:⏸ }${percent-pos}% [${time-pos}/${duration}] ${filename} ${?pause==yes:(paused)}"; cycle vid'
```

I noticed it when fiddling with the playlist shown by the OSD. After going back from Sixel (or after cycling back, don't know which is the cause), nothing gets refreshed in the text output anymore.

**Reply by vtmx** [reply]
Your help is very userfull, finally I can cleaner the output with config in: `~/.config/mpv/mpv.conf`. Thanks!

```
msg-level=all=error,statusline=status
term-status-msg='\n${filename}\n${metadata/title}\n${metadata/artist}\n${time-pos} / ${duration} (${percent-pos}%) ${?pause==yes:(paused)}'
```
